### PR TITLE
Add error message for wrong number of argument patterns

### PR DESCRIPTION
**Pull request by gosubpl**
Refs #1589

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 1]
```diff
@@ -1929,4 +1929,16 @@ object messages {
...
+  case class UnapplyInvalidNumberOfArguments(qual: untpd.Tree, argTypes: List[Type])(implicit ctx: Context)
+    extends Message(UnapplyInvalidNumberOfArgumentsID) {
+    val kind = "Syntax"
+    val msg = hl"Wrong number of argument patterns for $qual; expected: ($argTypes%, %)"
+    val explanation =
+      hl"""The Unapply method of $qual was used with incorrect number of arguments.
+          |Expected usage would be something like:
+          |case $qual(${argTypes.map(_ => '_')}%, %) => ...
+          |
+        |where subsequent arguments would have following types: ($argTypes%, %).""".stripMargin
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: `""".stripMargin` should be on the next line.

**Comment by gosubpl** [comment]
@nicolasstucki thanks for such a quick review! I have reflected your comments.

File: compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala (hunk 1 above)
Comment (allanrenucci) [reply compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: The issue with `stripMargin` being on the next line is that it adds an extra line to the explanation which is not desired.

Comment (gosubpl) [reply compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: @nicolasstucki @allanrenucci can you please decide which option is recommended? I will gladly oblige.
Also - whoever does the merge can fix it as I have checked 'Allow edits from maintainers'.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: Use Allan's.

**Comment by gosubpl** [comment]
Many thanks!

**Pull request merged**